pkg/orchestrator/cache.go

```diff
@@ -20,23 +22,118 @@ const (
 	defaultPurge  = 30 * time.Second
 )
 
+// TODO: make it possible to find information about single workloads
+
+var (
+	CacheExpVars        = expvar.NewMap("orchestrator-cache")
+	DeploymentCacheHits = expvar.Int{}
+	ReplicaSetCacheHits = expvar.Int{}
+	NodeCacheHits       = expvar.Int{}
+	ServiceCacheHits    = expvar.Int{}
+	PodCacheHits        = expvar.Int{}
+
+	SendExpVars    = expvar.NewMap("orchestrator-sends")
+	DeploymentHits = expvar.Int{}
+	ReplicaSetHits = expvar.Int{}
+	NodeHits       = expvar.Int{}
+	ServiceHits    = expvar.Int{}
+	PodHits        = expvar.Int{}
+)
+
+func init() {
+	CacheExpVars.Set("PodsCacheHits", &PodCacheHits)
+	CacheExpVars.Set("DeploymentCacheHits", &DeploymentCacheHits)
+	CacheExpVars.Set("ReplicaSetsCacheHits", &ReplicaSetCacheHits)
+	CacheExpVars.Set("NodeCacheHits", &NodeCacheHits)
+	CacheExpVars.Set("ServiceCacheHits", &ServiceCacheHits)
+
+	SendExpVars.Set("Pods", &PodHits)
+	SendExpVars.Set("Deployment", &DeploymentHits)
+	SendExpVars.Set("ReplicaSets", &ReplicaSetHits)
+	SendExpVars.Set("Node", &NodeHits)
+	SendExpVars.Set("Service", &ServiceHits)
+}
+
 // KubernetesResourceCache provides an in-memory key:value store similar to memcached for kubernetes resources.
-var KubernetesResourceCache = cache.New(defaultExpire, defaultPurge)
+var (
+	KubernetesResourceDeploymentCache = cache.New(defaultExpire, defaultPurge)
+	KubernetesResourceReplicaSetCache = cache.New(defaultExpire, defaultPurge)
+	KubernetesResourcePodCache        = cache.New(defaultExpire, defaultPurge)
+	KubernetesResourceNodeCache       = cache.New(defaultExpire, defaultPurge)
+	KubernetesResourceServiceCache    = cache.New(defaultExpire, defaultPurge)
+)
 
 // SkipKubernetesResource checks with a global kubernetes cache whether the resource was already reported.
 // It will return true in case the UID is in the cache and the resourceVersion did not change. Else it will return false.
 // 0 == defaultDuration
-func SkipKubernetesResource(uid types.UID, resourceVersion string) bool {
+func SkipKubernetesResource(uid types.UID, resourceVersion string, nodeType NodeType) bool {
 	cacheKey := string(uid)
-	value, hit := KubernetesResourceCache.Get(cacheKey)
+	kubernetesResourceCache := getNodeTypeCache(nodeType)
+	value, hit := kubernetesResourceCache.Get(cacheKey)
 
 	if !hit {
-		KubernetesResourceCache.Set(cacheKey, resourceVersion, 0)
+		kubernetesResourceCache.Set(cacheKey, resourceVersion, 0)
+		incCacheMiss(nodeType)
 		return false
-	}
-	if value != resourceVersion {
-		KubernetesResourceCache.Set(cacheKey, resourceVersion, 0)
+	} else if value != resourceVersion {
+		incCacheMiss(nodeType)
+		kubernetesResourceCache.Set(cacheKey, resourceVersion, 0)
 		return false
+	} else {
+		incCacheHit(nodeType)
+		return true
+	}
+}
+
+// TODO: Refactor orchestrator related packaged to reduce those switch cases by using interfaces and types.
```

Comment:
It would be way nicer to refactor orchestrator related topics to use interfaces and proper types. But IMO this is something for a later PR as this touches a lot of different orchestrator related files and packages